components/data_prefetch/data_prefetch.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PQueue from 'p-queue';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Constants} from 'utils/constants';
+import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
+
+const queue = new PQueue({concurrency: 2});
+
+type Props = {
+    currentChannelId: string;
+    prefetchQueueObj: Record<string, string[]>;
+    prefetchRequestStatus: Record<string, any>;
+    unreadChannels: Channel[];
+    actions: {
+        prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<any>;
```

Comment:
I think this can be specified as
```suggestion
        prefetchChannelPosts: (channelId: string, delay?: number) => Promise<any>;
```
since delay is optional.